x/wasm-storage/keeper/abci.go

```diff
@@ -32,3 +64,94 @@ func (k Keeper) ProcessExpiredWasms(ctx sdk.Context) error {
 	}
 	return nil
 }
+
+func (k Keeper) ExecuteTally(ctx sdk.Context) error {
+	// 1. Get contract address.
+	contractAddr, err := k.ProxyContractRegistry.Get(ctx)
+	if contractAddr == "" || errors.Is(err, collections.ErrNotFound) {
+		k.Logger(ctx).Debug("proxy contract address not registered")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// 2. Fetch tally-ready DRs.
+	queryRes, err := k.wasmViewKeeper.QuerySmart(ctx, sdk.MustAccAddressFromBech32(contractAddr), []byte(`{"get_data_requests_by_status":{"status": "tallying"}}`))
```

Comment:
`MustAccAddressFromBech32` leaves some chance for panic. Better check for error while getting address from Bech32.

Reply:
this is addressed in `hy/exec`